### PR TITLE
CUDA-enabled torchvision/v0.16.0

### PR DESCRIPTION
**Pull request by moravveji**
This PR adds a new easyconfig file for building `torchvision/0.16.0-foss-2023a-CUDA-12.1.1` and complements a similar easyconfig file for a CPU-only build.

### CODE REVIEW

File: easybuild/easyconfigs/p/PyTorch/PyTorch-2.1.2-foss-2023a-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -171,7 +171,7 @@ dependencies = [
     ('expecttest', '0.1.5'),
     ('networkx', '3.1'),
     ('sympy', '1.12'),
-    ('Z3', '4.12.2'),
+    ('Z3', '4.12.2', '-Python-%(pyver)s', ('GCCcore', '12.3.0')),
```
Comment (casparvl) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-2.1.2-foss-2023a-CUDA-12.1.1.eb:174]: This was taken out on purpose, see https://github.com/easybuilders/easybuild-easyconfigs/pull/20050
We now have _one_ `Z3` which has Python support. Ie. no more build with/without suffix. Please undo this change. You can then build your `torchvision` against PR 20050.

**Comment by VRehnberg** [comment]
This will probably build without JPEG support and is probably not running the torchvision test-suite. Compare against https://github.com/easybuilders/easybuild-easyconfigs/pull/19987/

**Review by Micket — CHANGES_REQUESTED** [review]
I strongly propose we shouldn't use these at all, but use the PyTorch-bundle's instead.
